### PR TITLE
[v2] ensure update() adheres to requiredNamespaces

### LINKED ISSUE #1121: Handle requiredNamespaces in update method

### PR DESCRIPTION
**Pull request by xzilja**
closes #1121

### CODE REVIEW

File: packages/sign-client/test/validation.spec.ts [hunk 1]
```diff
@@ -369,6 +371,21 @@ describe("Sign Client Validation", () => {
         "Missing or invalid update namespaces",
       );
     });
+
+    it("thorws when incompatible namespaces params are provided", async () => {
```
Comment (bkrem) [inline packages/sign-client/test/validation.spec.ts]: Typo:
```suggestion
    it("throws when incompatible namespaces params are provided", async () => {
```

File: packages/utils/src/validators.ts [hunk 2]
```diff
@@ -286,3 +286,10 @@ export function isValidNamespacesEvent(
   const events = getNamespacesEventsForChainId(namespaces, chainId);
   return events.includes(eventName);
 }
+
+export function isValidNamespacesChange(
+  requiredNamespaces: ProposalTypes.RequiredNamespaces,
+  namespaces: SessionTypes.Namespaces,
+) {
+  return !Object.keys(getDeletedDiff(requiredNamespaces, namespaces)).length;
```
Comment (bkrem) [inline packages/utils/src/validators.ts]: I think the simpler semantics would be to avoid the negation and explicitly do `Object.keys(...).length === 0`

**Review by bkrem — APPROVED** [review]
Nice! 💯 

Took my time to go over `getDeletedDiff` a few times because of `.reduce` + recursion but looks good to me 👍

**Comment by xzilja** [comment]
> Took my time to go over getDeletedDiff a few times because of .reduce + recursion but looks good to me 👍

@bkrem yeh, implementation is adapted from https://www.npmjs.com/package/deep-object-diff but since we only needed that small bit, I though its best to make a small util instead of using lib

**Comment by bkrem** [comment]
Ahh ok thanks for the link to that 👍 Yeah agreed, if we just need a small subset or an adaption of it pulling in the whole dep is overkill.

**Pull request merged**